promql/engine.go

```diff
@@ -2867,28 +2867,61 @@ func (ev *evaluator) aggregation(e *parser.AggregateExpr, q float64, inputMatrix
 		group := &groups[seriesToResult[si]]
 		// Initialize this group if it's the first time we've seen it.
 		if !group.seen {
-			*group = groupedAggregation{
-				seen:                   true,
-				floatValue:             f,
-				floatMean:              f,
-				incompatibleHistograms: false,
-				groupCount:             1,
-			}
 			switch op {
 			case parser.AVG, parser.SUM:
+				*group = groupedAggregation{
+					seen:                   true,
+					floatValue:             f,
+					floatMean:              f,
+					incompatibleHistograms: false,
+					groupCount:             1,
+				}
 				if h == nil {
 					group.hasFloat = true
 				} else {
 					group.histogramValue = h.Copy()
 					group.hasHistogram = true
 				}
 			case parser.STDVAR, parser.STDDEV:
-				group.floatValue = 0
+				if h == nil {
```

Comment:
Instead of initialising the groupedAggregation for each case, can we just set the `group.seen=false` when histogram is encountered for the first time in the group? If YES, then it will reduce a lot of code repeatations.

Reply:
I think this works. I have made some changes and pushed it to @jhesketh's branch. @jhesketh WDYT?